### DISCUSSION TITLE
Sequencing IDs for Elements in JointJS Plus Chatbot Application

### DISCUSSION THREAD

**Discussion by VinodKumarM7** [clientIO/joint#2594] [Q&A]
### Introduction

Dear all,

I'm currently working on a project that involves JointJS integrated with a chatbot application. In this application, I'm faced with the task of updating the IDs of elements or shapes within JointJS. However, I need these IDs to follow a sequential order rather than being randomly generated.

To elaborate further, I require assistance on how to implement a mechanism that ensures the IDs assigned to elements or shapes are sequential and can be updated accordingly within the JointJS framework.

Could anyone provide insights, suggestions, or perhaps even code examples on how to achieve this? I would greatly appreciate any guidance or pointers that could help me accomplish this task effectively.

Thank you in advance for your time and assistance.

Best regards,
VK

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
There is the `generateId()` method that can be overridden in shape definition. See the `dia.Cell` [source code](https://github.com/clientIO/joint/blob/master/packages/joint-core/src/dia/Cell.mjs#L141-L143).

```ts
const MyElement = dia.Element.define('MyElement', {
   // attributes
}, {
  generateId() {
     return myGenerator.nextId();
  }
});
```

But you can also introduce another attribute e.g. `appId` and use it as a unique identifier across your application.

```ts
element.set('appId', myGenerator.nextId());
```
